src/main/java/de/tum/in/www1/artemis/web/rest/AssessmentResource.java

```diff
@@ -59,8 +59,13 @@ void checkAuthorization(Exercise exercise, User user) throws AccessForbiddenExce
 
     void validateExercise(Exercise exercise) throws BadRequestAlertException {
         Course course = exercise.getCourse();
-        if (course == null) {
-            throw new BadRequestAlertException("The course belonging to this exercise does not exist", getEntityName(), "courseNotFound");
+        try {
+            if (course == null && exercise.getExerciseGroup().getExam().getCourse() == null) {
+                throw new BadRequestAlertException("The course belonging to this exercise does not exist", getEntityName(), "courseNotFound");
+            }
+        }
+        catch (NullPointerException nullPointerException) {
+            throw new BadRequestAlertException("The course belonging to this exercise or its exercise group and exam does not exist", getEntityName(), "courseNotFound");
```

Comment:
Don't you already have a method to verify the course of an exercise? Not sure were it's located maybe someone else can help here?

Reply:
This method is just refactored, maybe it is also done a PR under review. In that case this would be resolved during the merge